index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
Are you somehow expecting to enforce a stronger type than JS does for your `Maybe`s?  Do you have somehow a `Maybe(String)` type different from a `Maybe([Number])`?

If not, I don't understand your `concat`.  How do you know that you can concat the `value` of this `Maybe` with the value of that `Maybe`?  What happens with `Maybe.of([1, 2, 3]).concat(Maybe.of("abc"));` or vice versa?